### PR TITLE
Parallel transforms

### PR DESCRIPTION
**Pull request by bigdaz**

### CODE REVIEW

File: subprojects/base-services/src/main/java/org/gradle/internal/operations/DefaultBuildOperationProcessor.java [hunk 1]
```diff
@@ -36,17 +36,28 @@
...
+    public DefaultBuildOperationProcessor(BuildOperationWorkerRegistry buildOperationWorkerRegistry, BuildOperationQueueFactory buildOperationQueueFactory, ExecutorFactory executorFactory, int maxWorkerCount) {
+        this.buildOperationWorkerRegistry = buildOperationWorkerRegistry;
         this.buildOperationQueueFactory = buildOperationQueueFactory;
         this.fixedSizePool = executorFactory.create("build operations", maxWorkerCount);
     }
 
     @Override
     public <T extends BuildOperation> void run(BuildOperationWorker<T> worker, Action<BuildOperationQueue<T>> generator) {
-        BuildOperationQueue<T> queue = buildOperationQueueFactory.create(fixedSizePool, worker);
+        BuildOperationWorkerRegistry.Completion completion = buildOperationWorkerRegistry.maybeStartOperation();
```
Comment (adammurdoch) [inline subprojects/base-services/src/main/java/org/gradle/internal/operations/DefaultBuildOperationProcessor.java]: This doesn't look right. There should always be a build operation associated with the current thread.

Comment (bigdaz) [inline subprojects/base-services/src/main/java/org/gradle/internal/operations/DefaultBuildOperationProcessor.java]: At this stage there isn't any `BuildOperationWorkerRegistry.Operation` registered during configuration. I thought I put a comment in about how this hack won't be necessary when we merge the operations in `BuildOperationExecuter` with those used in `BuildOperationWorkerRegistry`.

**Comment by bigdaz** [comment]
At this stage I have about 15 different not-quite-right implementations of parallel transformations. This branch is the common stuff from those.

File: subprojects/base-services/src/main/java/org/gradle/internal/operations/DefaultBuildOperationProcessor.java (hunk 1 above)
Comment (adammurdoch) [inline subprojects/base-services/src/main/java/org/gradle/internal/operations/DefaultBuildOperationProcessor.java]: ok. I thought we had linked them up already. I guess not

**Pull request merged**